internal/clients/elasticsearch/index.go

```diff
@@ -208,15 +210,33 @@ func DeleteIndexTemplate(ctx context.Context, apiClient *clients.ApiClient, temp
 	return diags
 }
 
-func PutIndex(ctx context.Context, apiClient *clients.ApiClient, index *models.Index) diag.Diagnostics {
+var includeTypeNameUnsupportedVersion = version.Must(version.NewVersion("8.0.0"))
```

Comment:
nit: IMO it would be nice to include if this is an upper or lower bound in the name. 

```suggestion
var includeTypeNameMinUnsupportedVersion = version.Must(version.NewVersion("8.0.0"))
```